### PR TITLE
restore minimal-bitrot-prevention minimal preset tests

### PR DESCRIPTION
**Pull request by tersec**
If this works on Azure 32-bit, ideally this much of the minimal preset remains tested. The rest of the preset follows directly, but ensures that aspect of the spec remains followed. The minimal preset SSZ objects, for example, are just direct consequences of these changes, parameterized through `datatypes.nim`.

By not pulling in all the dependencies that `all_tests` or similar would, and having negligible runtime, this should avoid the 32-bit Azure timeouts.

### CODE REVIEW

File: tests/official/test_fixture_const_sanity_check.nim [hunk 1]
```diff
@@ -83,9 +83,15 @@ macro parseNumConsts(file: static string): untyped =
 const datatypesConsts = @(parseNumConsts(SpecDir/"datatypes.nim"))
 
 when const_preset == "minimal":
-  const minimalConsts = @(parseNumConsts(SpecDir/"presets"/"minimal.nim"))
+  when ETH2_SPEC == "v0.12.1":
```
Comment (arnetheduck) [inline tests/official/test_fixture_const_sanity_check.nim:86]: shall we move the presets to `presets/version/xxx.nim`?

Comment (tersec) [reply tests/official/test_fixture_const_sanity_check.nim:86]: This whole setup is temporary, gone in a week or two in all likelihood.

Comment (arnetheduck) [reply tests/official/test_fixture_const_sanity_check.nim:86]: there will be a 1.0, and a 1.1, and...

Comment (tersec) [reply tests/official/test_fixture_const_sanity_check.nim]: https://github.com/status-im/nim-beacon-chain/pull/1202/commits/cd17f5c4ddcb188cf8e4bd49901b28f5a5b04239

**Comment by tersec** [comment]
Closing PR, prettier not to deal with rebasing when it's sort of wholesale changes

**Comment by tersec** [comment]
And https://github.com/status-im/nim-beacon-chain/pull/1187 is just going to hit this again if I do that, or vice/versa, so this can wait for that.

**Pull request closed without merging**